src/Cards/Cards.js

```diff
@@ -14,12 +15,13 @@ export default class Cards extends Component {
   }
 
   async componentDidMount() {
+    //make a new request for all  movie details and create an array of all those movies
```

Comment:
You're grabbing all of the full movie details at once when the component mounts?